### ISSUE TITLE
"404 Error [Go back home]" screen appeared in place of "contentBuilder preview pane" 

### ISSUE THREAD

**Issue by kovacsgergelydpr** [pruvious/pruvious#33]
node v20.9.0
npm 10.1.0
pruvious 3.11.7
nuxt 3.11.2

I have created a "non-pageLike" but "contentBuilder  truthy" collection and when I clicked the add button, a "404 Error [Go back home]" screen appeared in place of "contentBuilder preview pane".

I see this request on the network tab (preview token is missing): 
```http://localhost:3000/?__p=_```

Please help, what am I doing wrong..

collections/tariffcards.ts:
```import { defineCollection } from '#pruvious'
import { uniqueValidator } from '#pruvious/server'

export default defineCollection({
    name: 'tariffcards',
    mode: 'multi',
    search: {
        default: [
            { field: 'name', reserve: 160 },
            { field: 'blocks', fieldValueType: 'populated' },
        ],
    },
    translatable: true,
    contentBuilder: {
        blocksField: 'blocks',
    },
    dashboard: {
        icon: 'Transform',
        primaryField: 'name',
        overviewTable: {
            columns: [{ field: 'name', width: 72.9 }, 'createdAt', 'updatedAt'],
        },
    },
    fields: {
        /*
        |--------------------------------------------------------------------------
        | name
        |--------------------------------------------------------------------------
        |
        */
        name: {
            type: 'text',
            options: {
                label: 'Name',
                required: true,
            },
            additional: {
                unique: 'perLanguage',
                validators: [
                    (context: any) =>
                        uniqueValidator(
                            context,
                            context.__(context.language, 'pruvious-server', 'A tariffcard with this name already exists'),
                        ),
                ],
            },
        },

        /*
        |--------------------------------------------------------------------------
        | blocks
        |--------------------------------------------------------------------------
        |
        */
        blocks: {
            type: 'repeater',
            options: {
                label: 'Blocks',
                description: 'The blocks that make up the preset content.',
                subfields: {
                    block: {
                        type: 'block',
                        options: {
                            label: 'Block',
                        },
                    },
                },
                addLabel: 'Add block',
            },
        },
    },
})
```

**Comment by murisceman** [comment]
To allow previews for a collection, it must have the `publicPages` setting enabled. The minimum required configuration is:

```ts
// ...
publicPages: {
  pathPrefix: '_tariffcards', // Some unlikely path
  pathField: 'path', // We need to mock this field
  publicField: 'public', // We set this to `false` to exclude this collection from the sitemap
},
fields: {
  path: {
    type: 'text',
    options: {},
    additional: {
      hidden: true, // Don't show in dashboard
      immutable: true, // Cannot be updated
      sanitizers: [{ onCreate: true, sanitizer: () => nanoid() }], // Randomly generated path
    },
  },
  public: {
    type: 'switch', // boolean
    options: {},
    additional: {
      hidden: true,
      immutable: true,
      sanitizers: [{ onCreate: true, sanitizer: () => false }], // Exclude from sitemap
    },
  },
  // ...
},
// ...
```

**Comment by kovacsgergelydpr** [comment]
It works seamlessly.
Thanks for the quick answer, keep up your amazing work!

**Comment by andreasvirkus** [comment]
Is this not available for `pageLikeCollections` @murisceman? Also getting a 404, seemingly because of the `?__p=` (this actually causes any page to return a 404)

**Comment by murisceman** [comment]
@andreasvirkus, it is available for page-like collections. The `__p` query parameter contains a token for previewing a non-public page record. Could you provide more details about the 404 error you're encountering, preferably in a new issue? 🙏
